### PR TITLE
some changes / fixes to pillwotop metrics

### PR DESCRIPTION
**Pull request by snopoke**
can review by commit :fish:

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/pillow/interface.py [hunk 1]
```diff
@@ -104,7 +104,9 @@ def process_changes(self, since, forever):
                         self._record_change_exception_in_datadog(change)
                         raise
                     else:
-                        self.fire_change_processed_event(change, context)
+                        updated = self.fire_change_processed_event(change, context)
+                        if updated:
+                            self._record_checkpoint_in_datadog()
```
Comment (snopoke) [inline corehq/ex-submodules/pillowtop/pillow/interface.py]: Not sure where you mean but most other places you might put this don't have all the context required.

Comment (emord) [inline corehq/ex-submodules/pillowtop/pillow/interface.py]: I meant in the event handler. Makes sense it wouldn't have enough context

File: corehq/ex-submodules/pillowtop/checkpoints/manager.py [hunk 2]
```diff
@@ -119,6 +119,9 @@ def update_checkpoint(self, new_seq):
     def fire_change_processed(self, change, context):
         if self.should_update_checkpoint(context):
             self.update_checkpoint(change['seq'])
+            return True
+
+        return False
```
Comment (benrudolph) [inline corehq/ex-submodules/pillowtop/checkpoints/manager.py]: 👍

**Review by benrudolph — APPROVED** [review]
nice, thanks for the cleanup

File: corehq/ex-submodules/pillowtop/feed/mock.py [hunk 3]
```diff
@@ -12,20 +12,25 @@ def __init__(self, queue):
...
         return len(self._queue)
 
-    def get_current_offsets(self):
-        return {'test': self.get_latest_change_id()}
+    def get_processed_offsets(self):
+        return {'test': self._since}
 
-    def get_checkpoint_value(self):
-        return str(self.get_latest_change_id())
+    def get_processed_offsets(self):
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/feed/mock.py]: F811 redefinition of unused 'get_processed_offsets' from line 29

File: corehq/ex-submodules/pillowtop/feed/mock.py [hunk 4]
```diff
@@ -38,23 +43,28 @@ class RandomChangeFeed(ChangeFeed):
...
         return self._count
 
-    def get_current_offsets(self):
-        return {'test': self.get_latest_change_id()}
+    def get_processed_offsets(self):
+        return {'test': self._since}
 
-    def get_checkpoint_value(self):
-        return str(self.get_latest_change_id())
+    def get_processed_offsets(self):
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/feed/mock.py]: F811 redefinition of unused 'get_processed_offsets' from line 63

**Comment by emord** [comment]
build failed. not obvious since the last commit uses [ci skip]
```
========================================
FAIL: test_checkpoint_with_multiple_topics (corehq.apps.change_feed.tests.test_kafka_change_feed.KafkaCheckpointTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/util/decorators.py", line 24, in decorated
    return fn(*args, **kwds)
  File "/mnt/commcare-hq-ro/corehq/apps/change_feed/tests/test_kafka_change_feed.py", line 97, in test_checkpoint_with_multiple_topics
    self.assertEqual(feed.get_current_checkpoint_offsets(), pillow.get_last_checkpoint_sequence())
AssertionError: {'case': 5, 'form': 2} != {u'case': 4, u'form': 2}
- {'case': 5, 'form': 2}
?          ^
+ {u'case': 4, u'form': 2}
?  +        ^  +
----------------------------------------------------------------------
```

**Comment by benrudolph** [comment]
tests passed, merging

**Pull request merged**